packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,122 @@
+import { Transaction } from "bitcoinjs-lib-v5";
+import {
+  unsignedMultisigTransaction, 
+  multisigRequiredSigners,
+  multisigTotalSigners,
+  multisigAddressType
+} from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { RBFOptions, UTXO, TransactionOutput, FeeRate } from "./types";
+import { isRBFSignaled, estimateVirtualSize } from "./utils";
+
+/**
+ * Prepare an RBF (Replace-By-Fee) transaction.
+ * This function doesn't broadcast the transaction, it just prepares it.
+ * @param options RBF options including the original transaction and new fee rate
+ * @returns A new transaction ready for signing and broadcasting
+ */
+export function prepareRBFTransaction(options: RBFOptions): Transaction {
+  const {
+    transaction,
+    newFeeRate,
+    utxos,
+    subtractFromOutput = false,
+    cancelTransaction = false,
+    destinationAddress,
+    network,
+  } = options;
+
+  if (!isRBFSignaled(transaction)) {
+    throw new Error("Original transaction is not signaling RBF");
+  }
+
+  // Calculate new fee
+  const newFee = calculateNewFee(transaction, newFeeRate);
```

Comment:
This makes me feel like a class could be a good option for this too. I imagine something like a `RbfTransaction` that could have a method `tx.getCanceled()` or `tx.getAccelerated`.

Reply:
now we have classes for both rbf and cpfp :)